SingularityUI/app/application.coffee

```diff
@@ -113,9 +113,17 @@ class Application
                     message:   "<p>Could not reach the Singularity API. Please make sure SingularityUI is properly set up.</p><p>If running through Brunch, this might be your browser blocking cross-domain requests.</p>"
             else
                 try
-                  serverMessage = JSON.parse(jqxhr.responseText).message or jqxhr.responseText
+                    serverMessage = JSON.parse(jqxhr.responseText).message or jqxhr.responseText
                 catch
-                  serverMessage = jqxhr.responseText
+                    if jqxhr.status is 200
+                        console.error jqxhr
+                        Messenger().error
+                            message:    '''
+                                            <p>Expected JSON but received something else (possibly html). The response has been saved to your js console.</p>
+                                            <p>One possible cause is an http redirect to an html web page.</p>
```

Comment:
I still think we can be a bit smarter about the messaging here. We can know if it is html by checking if the string starts with `<!DOCTYPE html>`. Then we can either say '...but received html' or '...but received something else' if we don't know.

Also, The one possible cause is very specific to our own use case. If there is no way for us to know that that might be the error, we shouldn't be mentioning possible causes.